src/PyListProxyHandler.cc

```diff
@@ -1244,7 +1243,7 @@ static uint32_t FlattenIntoArray(JSContext *cx,
       elementVal.set(jsTypeFactory(cx, PyList_GetItem(source, sourceIndex)));
     }
 
-    PyObject *element = pyTypeFactory(cx, elementVal)->getPyObject();
+    PyObject *element = pyTypeFactory(cx, elementVal);
```

Comment:
`element` needs to be `Py_DECREF`'d before the end of this loop iteration

Reply:
before end of loop actually